### ISSUE TITLE
matplotlib qt backend shows figure very slowly in jupyterlab. 

### ISSUE THREAD

**Issue by jiadong93** [matplotlib/matplotlib#18031]
### Bug report

**Bug summary**

In jupyterlab, I use the following code to plot figure, which is quite slow to show, pls see attached gif. When I swithch to `tk`, it is fast to show, but if there is error in the cell, the kernel will restarted automatically [this is another issue though]. 

**Code for reproduction**

```python
%matplotlib qt
import numpy as np
import matplotlib.pyplot as plt

# Data for plotting
t = np.arange(0.0, 2.0, 0.01)
s = 1 + np.sin(2 * np.pi * t)

fig, ax = plt.subplots()
ax.plot(t, s)

ax.set(xlabel='time (s)', ylabel='voltage (mV)',
       title='About as simple as it gets, folks')
ax.grid()
```

**Actual outcome**

![qt_backend](https://user-images.githubusercontent.com/15790176/88275130-6de2b900-cd0f-11ea-99b6-e4bfc6483b5b.gif)

**Expected outcome**

It should not have too much delay. I use the same desktop and qt backend, but never experienced such a long delay before.

**Matplotlib version**

  * Operating system: Windows 10
  * Matplotlib version:  3.2.2
  * Matplotlib backend (`print(matplotlib.get_backend())`): Qt5Agg
  * Python version: 3.7.7
  * Jupyter version (if applicable):
jupyter core     : 4.6.3
jupyter-notebook : 6.0.3
qtconsole        : not installed
ipython          : 7.16.1
ipykernel        : 5.3.3
jupyter client   : 6.1.6
jupyter lab      : 2.1.5
nbconvert        : 5.6.1
ipywidgets       : 7.5.1
nbformat         : 5.0.7
traitlets        : 4.3.3
  * Other libraries: numpy 1.18.5

**Comment by tacaswell** [comment]
attn @jasongrout I suspect this has to do with how the GUI event loops are integrated with kernel event loop on windows.

For more details about what is going on behind the curtain here see https://matplotlib.org/users/interactive_guide.html

**Comment by jiadong93** [comment]
Hi, @tacaswell , @dstansby , thanks for your replys. Notice the tk backend also got some issues. I can provide more details. This is definitly not my desktop perfeormance issue. As I downloaded the defalut version of anaconda env, lower version of matplotlib and Jupyterlab, all just went back to normal. Below are addtional info.

  * Operating system: Windows 10
  * Matplotlib version:  '3.1.3'
  * Matplotlib backend (`print(matplotlib.get_backend())`): Qt5Agg
  * Python version: 3.7.6
  * Jupyter version (if applicable):
jupyter core     : 4.6.1
jupyter-notebook : 6.0.3
qtconsole        : 4.6.0
ipython          : 7.12.0
ipykernel        : 5.1.4
jupyter client   : 5.3.4
jupyter lab      : 1.2.6 (this seems to work)
nbconvert        : 5.6.1
ipywidgets       : 7.5.1
nbformat         : 5.0.4
traitlets        : 4.3.3
  * Other libraries: numpy 1.18.1

**Comment by jiadong93** [comment]
> attn @jasongrout I suspect this has to do with how the GUI event loops are integrated with kernel event loop on windows.
> 
> For more details about what is going on behind the curtain here see https://matplotlib.org/users/interactive_guide.html

@jasongrout, pls see additional details, it seems loweing jupyterlab version solved the issue, this might a Jupyterlab issue also. Thanks for your attention:)

**Comment by blink1073** [comment]
`ipkykernel 5.3.3` had a bug that was fixed in `5.3.4`, cf https://github.com/ipython/ipykernel/pull/531

`pip install -U ipykernel` should fix the issue.

**Comment by timhoffm** [comment]
Assuming this is closed via ipykernel 5.3.4.

Please report back if the problem continues to exist.
